core/src/main/java/org/apache/gravitino/metalake/MetalakeManager.java

```diff
@@ -253,22 +295,24 @@ public BaseMetalake alterMetalake(NameIdentifier ident, MetalakeChange... change
               throw new MetalakeNotInUseException(
                   "Metalake %s is not in use, please enable it first", ident);
             }
-
-            return store.update(
-                ident,
-                BaseMetalake.class,
-                EntityType.METALAKE,
-                metalake -> {
-                  BaseMetalake.Builder builder = newMetalakeBuilder(metalake);
-                  Map<String, String> newProps =
-                      metalake.properties() == null
-                          ? Maps.newHashMap()
-                          : Maps.newHashMap(metalake.properties());
-                  builder = updateEntity(builder, newProps, changes);
-
-                  return builder.build();
-                });
-
+            METALAKE_CACHE.invalidate(ident);
+            BaseMetalake baseMetalake =
+                store.update(
+                    ident,
+                    BaseMetalake.class,
+                    EntityType.METALAKE,
+                    metalake -> {
+                      BaseMetalake.Builder builder = newMetalakeBuilder(metalake);
+                      Map<String, String> newProps =
+                          metalake.properties() == null
+                              ? Maps.newHashMap()
+                              : Maps.newHashMap(metalake.properties());
+                      builder = updateEntity(builder, newProps, changes);
+
+                      return builder.build();
+                    });
+            METALAKE_CACHE.put(ident, newMetalakeWithResolvedProperties(baseMetalake));
```

Comment:
here should use ident  of `baseMetalake`

Reply:
I see, good catch